pallets/manta-sbt/src/lib.rs

```diff
@@ -284,6 +351,116 @@ pub mod pallet {
+        ///
+        /// Requires a valid `Eip712Signature` which is generated from signing the zkp with an eth private key
+        #[pallet::call_index(3)]
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::mint_sbt_eth())]
+        #[transactional]
+        pub fn mint_sbt_eth(
+            origin: OriginFor<T>,
+            post: Box<TransferPost>,
+            eth_signature: Eip712Signature,
+            address_type: EvmAddressType,
+            metadata: BoundedVec<u8, T::SbtMetadataBound>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            // check that mint type is within time window
+            Self::check_mint_time(address_type.into())?;
+
+            let address = Self::verify_eip712_signature(&post.proof, &eth_signature)
+                .ok_or(Error::<T>::BadSignature)?;
+            // Check signature
+            match address_type {
+                EvmAddressType::Bab(eth_address) | EvmAddressType::Galxe(eth_address) => {
+                    ensure!(eth_address == address, Error::<T>::BadSignature);
+                }
+            }
+
+            let mint_status =
+                EvmAddressAllowlist::<T>::get(address_type).ok_or(Error::<T>::NotAllowlisted)?;
+            let asset_id = match mint_status {
+                MintStatus::Available(asset) => asset,
+                MintStatus::AlreadyMinted => return Err(Error::<T>::AlreadyMinted.into()),
+            };
+            // Change status to minted
+            EvmAddressAllowlist::<T>::insert(address_type, MintStatus::AlreadyMinted);
+
+            Self::check_post_shape(&post, asset_id)?;
+            SbtMetadata::<T>::insert(asset_id, metadata);
+
+            Self::post_transaction(vec![who], *post)?;
+            Self::deposit_event(Event::<T>::MintSbtEvm {
+                asset_id,
+                address: address_type,
+            });
+            Ok(().into())
+        }
+
+        /// Sets the privileged allowlist account. Requires `AdminOrigin`
+        #[pallet::call_index(4)]
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::change_allowlist_account())]
+        #[transactional]
+        pub fn change_allowlist_account(
+            origin: OriginFor<T>,
+            account: Option<T::AccountId>,
+        ) -> DispatchResult {
+            T::AdminOrigin::ensure_origin(origin)?;
+
+            AllowlistAccount::<T>::set(account.clone());
+            Self::deposit_event(Event::<T>::ChangeAllowlistAccount { account });
+            Ok(())
+        }
+
+        /// Sets the time range of which a `MintType` will be valid. Requires `AdminOrigin`
+        #[pallet::call_index(5)]
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::set_mint_time())]
+        #[transactional]
+        pub fn set_mint_time(
+            origin: OriginFor<T>,
+            mint_type: MintType,
+            start_time: T::Moment,
+            end_time: Option<T::Moment>,
```

Comment:
Do we need to add one line of code for checking ` if end_time.is_some() { end_time > start_time }` though it's a governance call?

Reply:
Yeah a check can be added, would just cause `mint_sbt_eth`  to always fail for that mint type. 

still an invariant that we can block from going on-chain